src/legacy/core_plugins/kibana/public/dashboard/index.js

```diff
@@ -57,7 +57,22 @@ function createNewDashboardCtrl($scope, i18n) {
 uiRoutes
   .defaults(/dashboard/, {
     requireDefaultIndex: true,
-    requireUICapability: 'dashboard.show'
+    requireUICapability: 'dashboard.show',
+    badge: (i18n, uiCapabilities) => {
+      if (uiCapabilities.dashboard.showWriteControls) {
+        return undefined;
+      }
+
+      return {
+        text: i18n('kbn.dashboard.badge.readOnly.text', {
+          defaultMessage: 'Read only',
+        }),
+        tooltip: i18n('kbn.dashboard.badge.readOnly.tooltip', {
+          defaultMessage: 'Unable to save dashboards',
+        }),
+        iconType: 'glasses'
```

Comment:
Would it make it easier to maintain consistency if the iconType and text were global vars?

Reply:
It definitely would. I played around with something similar [here](https://github.com/elastic/kibana/pull/34551/commits/c0e341bee1cea5c5a42ebc4d4efedc8abed586b1) but it was later reverted because it added some indirection.